corehq/util/spreadsheets_v2/adapters/generic.py

```diff
@@ -0,0 +1,17 @@
+from contextlib import contextmanager
+from corehq.util.spreadsheets_v2 import SpreadsheetFileExtError
+from .xls import open_xls_workbook
+from .xlsx import open_xlsx_workbook
+
+
+@contextmanager
+def open_any_workbook(filename):
+    if filename.endswith('.xls'):
+        with open_xls_workbook(filename) as workbook:
+            yield workbook
+    elif filename.endswith('.xlsx'):
+        with open_xlsx_workbook(filename) as workbook:
+            yield workbook
+    else:
+        raise SpreadsheetFileExtError('File {} does not end in .xls or .xlsx'
```

Comment:
Nope, I don't think so. These errors are all caught in the client code (i.e. case importer) and converted to error messages to be shown to the user. I'm generally opposed to the practice of showing an exception message to users as a sort of blanket ban.

This particular Exception never gets explicitly caught in case importer, but the xls/xlsx is explicitly checked for before this gets called, and if that ever changes this would 500 and still not be shown to the user.